modules/canbus/vehicle/ge3/ge3_controller.cc

```diff
@@ -0,0 +1,822 @@
+    // if the auto mode can be set normally, the harzad lamp should be off.
+    pc_bcm_201_->set_pc_hazardlampreq(Pc_bcm_201::PC_HAZARDLAMPREQ_NOREQ);
+    AINFO << "Switch to AUTO_SPEED_ONLY mode ok.";
+    return ErrorCode::OK;
+  }
+}
+
+// NEUTRAL, REVERSE, DRIVE
+void Ge3Controller::Gear(Chassis::GearPosition gear_position) {
+  if (!(driving_mode() == Chassis::COMPLETE_AUTO_DRIVE ||
+        driving_mode() == Chassis::AUTO_SPEED_ONLY)) {
+    AINFO << "this drive mode no need to set gear.";
+    return;
+  }
+  switch (gear_position) {
+    case Chassis::GEAR_NEUTRAL: {
+      pc_vcu_205_->set_pc_gearreq(Pc_vcu_205::PC_GEARREQ_NEUTRAL);
+      break;
+    }
+    case Chassis::GEAR_REVERSE: {
+      pc_vcu_205_->set_pc_gearreq(Pc_vcu_205::PC_GEARREQ_REVERSE);
+      break;
+    }
+    case Chassis::GEAR_DRIVE: {
+      pc_vcu_205_->set_pc_gearreq(Pc_vcu_205::PC_GEARREQ_DRIVE);
+      break;
+    }
+    case Chassis::GEAR_PARKING: {
+      pc_vcu_205_->set_pc_gearreq(Pc_vcu_205::PC_GEARREQ_PARK);
+      break;
+    }
+    case Chassis::GEAR_LOW: {
+      pc_vcu_205_->set_pc_gearreq(Pc_vcu_205::PC_GEARREQ_INVALID);
+      break;
+    }
+    case Chassis::GEAR_NONE: {
+      pc_vcu_205_->set_pc_gearreq(Pc_vcu_205::PC_GEARREQ_INVALID);
+      break;
+    }
+    case Chassis::GEAR_INVALID: {
+      AERROR << "Gear command is invalid!";
+      pc_vcu_205_->set_pc_gearreq(Pc_vcu_205::PC_GEARREQ_INVALID);
+      break;
+    }
+    default: {
+      pc_vcu_205_->set_pc_gearreq(Pc_vcu_205::PC_GEARREQ_INVALID);
+      break;
+    }
+  }
+}
+
+// brake with new acceleration
+// acceleration:0.00~99.99, unit:
+// acceleration:0.0 ~ 7.0, unit:m/s^2
+// acceleration_spd:60 ~ 100, suggest: 90
+// -> pedal
+void Ge3Controller::Brake(double pedal) {
+  // Update brake value based on mode
+  if (!(driving_mode() == Chassis::COMPLETE_AUTO_DRIVE ||
+        driving_mode() == Chassis::AUTO_SPEED_ONLY)) {
+    AINFO << "The current drive mode does not need to set acceleration.";
+    return;
+  }
+  pc_bcs_202_->set_pc_brkpedreq(pedal);
+}
+
+// drive with old acceleration
+// gas:0.00~99.99 unit:
+void Ge3Controller::Throttle(double pedal) {
+  if (!(driving_mode() == Chassis::COMPLETE_AUTO_DRIVE ||
+        driving_mode() == Chassis::AUTO_SPEED_ONLY)) {
+    AINFO << "The current drive mode does not need to set acceleration.";
+    return;
+  }
+  pc_vcu_205_->set_pc_accpedreq(pedal);
+}
+
+// ge3 default, -470 ~ 470, left:+, right:-
+// need to be compatible with control module, so reverse
+// steering with old angle speed
+// angle:-99.99~0.00~99.99, unit:, left:-, right:+
+void Ge3Controller::Steer(double angle) {
+  if (!(driving_mode() == Chassis::COMPLETE_AUTO_DRIVE ||
+        driving_mode() == Chassis::AUTO_STEER_ONLY)) {
+    AINFO << "The current driving mode does not need to set steer.";
+    return;
+  }
+  const double real_angle =
+      vehicle_params_.max_steer_angle() / M_PI * 180 * angle / 100.0;
```

Comment:
Suggest that you put the parameter `vehicle_params_.max_steer_angle() ` to configuration file.